### PR TITLE
iOS test app, fixed tests, disabled tests

### PR DESCRIPTION
**Pull request by orangemocha**
This adds:

1) an iOS app that runs the JXcore `simple` unit tests. 
2) Fixes to adapt a few tests to run on iOS.
3) Disabling of tests that are consistently failing or flaky on iOS.

---
This change is [Reviewable](https://reviewable.io/reviews/thaliproject/jxcore/96)

### CODE REVIEW

File: test/simple/test-http-many-keep-alive-connections.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ var common = require('../common');
 var assert = require('assert');
 var http = require('http');
 
-var expected = 10000;
+var expected = (process.platform === 'ios') ? 500 : 10000;
```
Comment (enricogior) [inline test/simple/test-http-many-keep-alive-connections.js]: This should be fixed for Android as well, 10000 connections don't make sense on a mobile device regardless of the OS.

**Comment by orangemocha** [comment]
We have decided that this tool is not ready for prime time, so it will not be merged yet. For now it will leave here https://github.com/thaliproject/jxcore/commits/orangemocha-ios-testapp

**Comment by yaronyg** [comment]
@orangemocha so if this isn't ready for prime time how do we propose to run the unit tests on iOS?

**Comment by orangemocha** [comment]
@yaronyg , this is not abandoned, just living in a separate branch. @enricogior felt that this is not land on master and 0.3.1.9 while it is still work in progress.

**Comment by enricogior** [comment]
@yaronyg the test app works and it allowed us to run the tests on iOS 10 and iOS 9, but with some limitations that I'm going to explain.
Given the jxcore tests have been designed to run on desktop with jx standalone, there are several tests we had to disable because, as they are, they cannot run on mobile. Beside the disabled tests, there are also tests that are currently flaky or failing and in some cases they may cause the following tests to fail as well, given that there isn't a full isolation between tests.
We don't have the time and the resources to fix all those tests right now (we fixed some but not all), but that doesn't prevent us from comparing the results of iOS 9 vs iOS 10. So we've ran the tests in batches (in some cases singularly) and verified that there aren't tests that pass on iOS 9 and fail on iOS 10, that is already a very important result.
But to do that comparison, we had to keep the flaky and the failing test as well, so if we commit the application to a stable branch you may think you can run it as it is, but that is not the case: you need to know how to run the batches and that has to be done manually. So we decided to leave it in a dedicated branch since it can't be used for example in a CI environment to run bv tests on iOS. The major point was to verify that we don't have regressions in iOS 10 and the app, in that regard, did its job.
If in the future, we will have the time and the resources to fix all the flaky and failing tests and improve the isolation between tests, we will be happy to push it to master.

**Comment by yaronyg** [comment]
@orangemocha @enricogior That makes a lot of sense. Thanks for explaining so we have a record.

**Pull request closed without merging**